### PR TITLE
feat: drawer footer standard props and some improvements

### PR DESCRIPTION
**Pull request by GiovannaMonti**
### Description

This PR follows the [Drawer component introduction PR](https://github.com/mia-platform/design-system/pull/494) and introduces the following changes:

#### Drawer component

**New features:**
- Drawer footer can now be a custom node or can leverage standard properties (button, extra) for a predefined look.
- Drawer now has a closeIcon
- Drawer can now display a doc link

**Improvements:**
- Drawer types have been moved to a separate file
- Drawer stories and mocks have been refactored to be more meaningful and usable; Drawer stories now correctly show the drawer props in the Show Code tab
- Drawer styles are now imported from a CSS module

### Checklist

- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [ ] typings are updated or integrated accordingly with your changes
- [ ] all added components are exported from index file (if necessary)
- [ ] all added files include Apache 2.0 license
- [x] you are not committing extraneous files or sensitive data
- [x] the browser console does not have any logged errors
- [x] necessary labels have been applied to this pull request (enhancement, bug, ecc.)

### CODE REVIEW

File: src/components/Drawer/Drawer.Footer.tsx [hunk 1]
```diff
@@ -16,14 +16,24 @@
...
-  footer: DrawerFooter,
-}
+export const Footer = ({ footer }: FooterProps): ReactElement | null => {
+  if (React.isValidElement(footer)) {
+    return footer
+  }
 
-export const Footer = ({ footer }: FooterProps): ReactElement => {
-  return <footer>{footer}</footer>
+  const drawerFooter = footer as DrawerFooter
```
Comment (fredmaggiowski) [inline src/components/Drawer/Drawer.Footer.tsx:29]: let's make a isDrawerFooter that returns the proper type casting instead of using `as`

File: src/components/Drawer/Drawer.test.tsx [hunk 2]
```diff
@@ -16,17 +16,20 @@
...
+import { DrawerLipsum, DrawerLipsumFooterButton, DrawerLipsumTitle } from './Drawer.mocks'
 import { render, screen } from '../../test-utils'
 import { Drawer } from './Drawer'
 import { DrawerProps } from './Drawer.props'
 
 describe('Drawer', () => {
   const props: DrawerProps = {
     children: 'Drawer Content',
-    footer: <DrawerLipsumFooter closeDrawer={jest.fn()} />,
+    footer: {
```
Comment (fredmaggiowski) [inline src/components/Drawer/Drawer.test.tsx:27]: we should test both footer types, here you have switched the test from on to the other prop type but both should be covered

File: src/components/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -39,15 +37,19 @@ export const Drawer = ({
   onClose,
   title,
 }: DrawerProps): ReactElement => {
+  const closeIcon = useMemo(() =>
```
Comment (fredmaggiowski) [inline src/components/Drawer/Drawer.tsx:40]: can this be instantiated outside the component?

File: src/components/Drawer/Drawer.test.tsx (hunk 2 above)
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.test.tsx:27]: yep was already onto it

File: src/components/Drawer/Drawer.tsx (hunk 3 above)
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.tsx:40]: I think so

File: src/components/Drawer/Drawer.tsx [hunk 4]
```diff
@@ -39,15 +37,19 @@ export const Drawer = ({
...
+  , [])
+
   return (
     <AntdDrawer
-      closeIcon={null}
+      className={styles.drawer}
+      closeIcon={closeIcon}
       destroyOnClose={destroyOnClose}
-      footer={<Drawer.Footer footer={footer} />}
+      footer={footer && <Drawer.Footer footer={footer} />}
```
[the remark this answers is not in the document]
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.tsx:49]: I did it this way because I wanted to avoid a Footer component that does not render anything

**Review by MIA-Deltat1995 — APPROVED** [review]

File: src/components/Button/Button.module.css [hunk 5]
```diff
@@ -21,6 +21,7 @@
   align-items: center;
   justify-content: center;
   gap: var(--spacing-gap-xs, 4px);
+  box-shadow: none;
```
Comment (fredmaggiowski) [inline src/components/Button/Button.module.css:24]: can you submit this change with a different PR?

the release note is generated with the PR titles and this is a relevant change that I'd not slip in the "footer standard props" pr :)

Comment (GiovannaMonti) [reply src/components/Button/Button.module.css:24]: oh, yes

File: src/components/Drawer/Drawer.Footer.tsx (hunk 1 above)
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.Footer.tsx:29]: I don't think there is a clean way to do that, I personally prefer to do it this way

File: src/components/Drawer/Drawer.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.tsx]: I changed it but it is now rendering the top border for some reason; need to fix it tomorrow

**Comment by epessina** [comment]
Can we consider adding a `docLink` prop to show the doc icon in the title like in the Modal?

File: src/components/Drawer/Drawer.Footer.tsx (hunk 1 above)
Comment (GiovannaMonti) [reply src/components/Drawer/Drawer.Footer.tsx]: done

**Comment by fredmaggiowski** [comment]
> Can we consider adding a `docLink` prop to show the doc icon in the title like in the Modal?

This makes sense, do other components have a standard prop definition for this? If so let's do this following the same prop-defnition, if not I'd open a PR to define such standard prop and then open another PR

**Review by fredmaggiowski — DISMISSED** [review]
LGTM

File: src/components/Drawer/Drawer.mocks.tsx [hunk 6]
```diff
@@ -40,33 +38,22 @@ export const DrawerLipsum = (): ReactElement => {
...
-        onClose={closeDrawer}
-      >
-        <DrawerLipsum />
-      </Drawer>
-    </>
-  )
+export const drawerLipsumFooter = {
+  buttons: [
+    <DrawerLipsumFooterButton key="action-button" text="Primary Action" />,
+    <DrawerLipsumFooterButton hierarchy={Hierarchy.Neutral} key="action-button" text="Secondary Action" />,
```
Comment (GiovannaMonti) [inline src/components/Drawer/Drawer.mocks.tsx:56]:
```suggestion
    <DrawerLipsumFooterButton hierarchy={Hierarchy.Neutral} key="secondary-action-button" text="Secondary Action" />,
```

**Comment by GiovannaMonti** [comment]
> Can we consider adding a `docLink` prop to show the doc icon in the title like in the Modal?



> > Can we consider adding a `docLink` prop to show the doc icon in the title like in the Modal?
> 
> This makes sense, do other components have a standard prop definition for this? If so let's do this following the same prop-defnition, if not I'd open a PR to define such standard prop and then open another PR

yes, the Modal has it. I can add it

**Review by fredmaggiowski — APPROVED** [review]
LGTM
